src/App.tsx

```diff
@@ -1,26 +1,142 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos, deleteTodo, addNewTodo } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/SortTypes';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState('');
+  const [idTodo, setIdTodo] = useState(0);
+  const [newTodo, setNewTodo] = useState({
+    userId: USER_ID,
+    title: '',
+    completed: false,
+  });
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        switch (status) {
+          case TodoStatus.Active:
+            setTodos(todosFromServer.filter(todo => !todo.completed));
+            break;
+          case TodoStatus.Completed:
+            setTodos(todosFromServer.filter(todo => todo.completed));
+            break;
+
+          default:
+            setTodos(todosFromServer);
+        }
+      })
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, [setTodos, status]);
+
+  function onDeleteTodo(todoId: number) {
+    setIdTodo(todoId);
+
+    return deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos => {
+          return currentTodos.filter(todo => todo.id !== todoId);
+        });
+      })
+      .catch(() => {
+        setTodos(todos);
+        setErrorMessage('Unable to delete a todo');
+      });
+  }
+
+  function onCreateTodo(newTodos: Omit<Todo, 'id'>) {
+    const trimmedTodo = { ...newTodos, title: newTodos.title.trim() };
+
+    setTodos(currentTodos => [...currentTodos, { ...newTodo, id: 0 }]);
+    setIdTodo(0);
+
+    return addNewTodo(trimmedTodo)
+      .then(todo => {
+        setTodos(todos);
+        setTodos(currentTodos => [...currentTodos, todo]);
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        setTodos(todos);
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)